### PR TITLE
Support user specified values when importing openapi def

### PR DESCRIPTION
**Pull request by psiinon**
Fixes zaproxy/zaproxy#3344

### CODE REVIEW

File: src/org/zaproxy/zap/extension/openapi/OpenApiSpider.java [hunk 1]
```diff
@@ -71,4 +75,14 @@ public boolean canParseResource(HttpMessage message, String path, boolean wasAlr
         log.debug("Cant parse " + message.getRequestHeader().getURI());
         return false;
     }
+    
+    private ValueGenerator getValueGenerator() {
+        if (this.valGen == null) {
+            ExtensionSpider spider = (ExtensionSpider) Control.getSingleton()
+                    .getExtensionLoader()
+                    .getExtension(ExtensionSpider.NAME);
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/openapi/OpenApiSpider.java]: If you use the `.class` variant you can avoid the cast.

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 2]
```diff
@@ -244,6 +245,17 @@ public void run() {
         return null;
     }
     
+    private ValueGenerator getValueGenerator() {
+        // Always get the latest ValueGenerator as it could have changed
+        ExtensionSpider spider = (ExtensionSpider) Control.getSingleton()
+                .getExtensionLoader()
+                .getExtension(ExtensionSpider.NAME);
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: If you use the `.class` variant you can avoid the cast.

Comment (psiinon) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: yeah, that would be better

**Comment by psiinon** [comment]
Oh, and I should update the help... will push those changes soon

File: src/org/zaproxy/zap/extension/openapi/generators/ValueGenerator.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Name : " + name + " Type : " + type + " Default : " + defaultValue + " Returning default value");
+            }
+            return defaultValue;
+        }
+        
+        HashMap<String, String> fieldAtts = new HashMap<String, String>();
+        fieldAtts.put("Control Type", type);
+        String value = coreValGen.getValue(null, null, name, defaultValue, 
+                new ArrayList<String>(),
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/generators/ValueGenerator.java]: Could use `Collections.emptyList()` instead (same for following statement, with `emptyMap`)?

File: test/org/zaproxy/zap/extension/openapi/OpenApiUnitTest.java [hunk 4]
```diff
@@ -177,14 +179,90 @@ Response serve(IHTTPSession session) {
         
         Requestor requestor = new Requestor(HttpSender.MANUAL_REQUEST_INITIATOR);
         HttpMessage defnMsg = this.getHttpMessage(test + "defn.yaml");
-        SwaggerConverter converter = new SwaggerConverter(requestor.getResponseBody(defnMsg.getRequestHeader().getURI()));
+        SwaggerConverter converter = new SwaggerConverter(requestor.getResponseBody(defnMsg.getRequestHeader().getURI()), null);
+        RequesterListener listener = new RequesterListener(){
+            @Override
+            public void handleMessage(HttpMessage message, int initiator) {
+            }};
+        requestor.addListener(listener);
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/openapi/OpenApiUnitTest.java]: Not needed?

**Comment by psiinon** [comment]
Hopefully fixed those comments.
Thanks for the v useful review (again;)

**Comment by psiinon** [comment]
Updated to add #3344 in the manifest

**Comment by psiinon** [comment]
Ugh, will look at the build failure now...

**Comment by psiinon** [comment]
Build ok now. I still need to fix a typo @thc202 just spotted.

For info I'm still hoping to implement zaproxy/zaproxy#3598 before releasing the add-on...

**Comment by psiinon** [comment]
Fixed typo and added example structured value. Ready to review again now.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**